lightning-invoice/src/payment.rs

```diff
@@ -557,6 +732,7 @@ where
 					self.scorer.lock().payment_path_failed(&path, *short_channel_id);
 				}
 
+				self.remove_path_inflight_htlcs(*payment_hash, path);
```

Comment:
We'll need to do this for the probing events, too, even though `InvoicePayer` hasn't exposed an interface for it yet. Let's pull these out into a separate `match` expression before this one to keep the code a bit more streamlined.

Here's a little rust pattern matching magic:

```rust
               match event {
                       Event::PaymentPathFailed { payment_hash, path, ..  }
                       | Event::PaymentPathSuccessful { path, payment_hash: Some(payment_hash), .. }
                       | Event::ProbeSuccessful { payment_hash, path, .. }
                       | Event::ProbeFailed { payment_hash, path, .. } => {
                               self.remove_path_inflight_htlcs(*payment_hash, path);
                       },
                       _ => {},
               }
```